terraform/containerinsight_eks/main.tf

```diff
@@ -0,0 +1,160 @@
+# so in the eks/k8s test, we need tester to provide the cluster instead of creating it in terraform
+# so that we can shorten the execution time
+
+terraform {
+  required_version = ">= 0.13"
+
+  required_providers {
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = ">= 1.7.0"
+    }
+  }
+}
+
+module "common" {
+  source = "../common"
+
+  aoc_image_repo = var.aoc_image_repo
+  aoc_version    = var.aoc_version
+}
+
+# region
+provider "aws" {
+  region = var.region
+}
+
+# get eks cluster by name
+data "aws_eks_cluster" "testing_cluster" {
+  name = var.eks_cluster_name
+}
+
+data "aws_eks_cluster_auth" "testing_cluster" {
+  name = var.eks_cluster_name
+}
+
+
+provider "kubectl" {
+  // Note: copy from eks module. Please avoid use shorted-lived tokens when running locally.
+  // For more information: https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#exec-plugins
+  host                   = data.aws_eks_cluster.testing_cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.testing_cluster.certificate_authority[0].data)
+  token                  = data.aws_eks_cluster_auth.testing_cluster.token
+  load_config_file       = false
+}
+
+data "template_file" "namespace_file" {
+  template = file("../testcases/containerinsight_eks/namespace.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+data "template_file" "service_account_file" {
+  template = file("../testcases/containerinsight_eks/service_account.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+data "template_file" "cluster_role_file" {
+  template = file("../testcases/containerinsight_eks/cluster_role.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+data "template_file" "cluster_role_binding_file" {
+  template = file("../testcases/containerinsight_eks/cluster_role_binding.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+data "template_file" "config_map_file" {
+  template = file("../testcases/containerinsight_eks/config_map.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+  }
+}
+
+data "template_file" "daemonset_file" {
+  template = file("../testcases/containerinsight_eks/daemonset.tpl")
+  vars = {
+    NAMESPACE = "aoc-ns-${module.common.testing_id}"
+    OTELIMAGE = module.common.aoc_image
+  }
+}
+
+resource "kubectl_manifest" "namespace" {
+  yaml_body = data.template_file.namespace_file.rendered
+}
+
+resource "kubectl_manifest" "service_account" {
+  yaml_body = data.template_file.service_account_file.rendered
+  depends_on = [
+    kubectl_manifest.namespace
+  ]
+}
+
+resource "kubectl_manifest" "cluster_role" {
+  yaml_body = data.template_file.cluster_role_file.rendered
+}
+
+resource "kubectl_manifest" "cluster_role_binding" {
+  yaml_body = data.template_file.cluster_role_binding_file.rendered
+  depends_on = [
+    kubectl_manifest.service_account,
```

Comment:
nit: ClusterRoleBinding doesn't depend on the creation of the object it binds to the ClusterRole. 
The service account is not created for controller, it's safe to remove the service_account from `depends_on`, it seems to be redundant.